presto-common/src/main/java/com/facebook/presto/common/type/TypeSignature.java

```diff
@@ -219,31 +222,36 @@ else if (c == ',') {
     private static class EnumMapParsingData
     {
         final int mapEndIndex;
+        private final String typeName;
         private final Map<String, String> map;
-        final boolean isLongEnum;
+        private final boolean isBigintEnum;
 
-        EnumMapParsingData(int mapEndIndex, Map<String, String> map, boolean isLongEnum)
+        EnumMapParsingData(int mapEndIndex, String typeName, Map<String, String> map, boolean isBigintEnum)
         {
             this.mapEndIndex = mapEndIndex;
+            this.typeName = typeName;
             this.map = map;
-            this.isLongEnum = isLongEnum;
+            this.isBigintEnum = isBigintEnum;
         }
 
         LongEnumMap getLongEnumMap()
         {
-            checkArgument(isLongEnum, "Invalid enum map format");
+            checkArgument(isBigintEnum, "Invalid enum map format");
             return new LongEnumMap(
+                    typeName,
                     map.entrySet().stream()
                             .collect(Collectors.toMap(Map.Entry::getKey, e -> Long.parseLong(e.getValue()))));
         }
 
         VarcharEnumMap getVarcharEnumMap()
         {
-            checkArgument(!isLongEnum, "Invalid enum map format");
+            checkArgument(!isBigintEnum, "Invalid enum map format");
             // Varchar enum values are base32-encoded so that they are case-insensitive, which is expected of TypeSigntures
             Base32 base32 = new Base32();
-            return new VarcharEnumMap(map.entrySet().stream()
-                    .collect(Collectors.toMap(Map.Entry::getKey, e -> new String(base32.decode(e.getValue().toUpperCase(ENGLISH))))));
+            return new VarcharEnumMap(
+                    typeName,
+                    map.entrySet().stream()
+                            .collect(Collectors.toMap(Map.Entry::getKey, e -> new String(base32.decode(e.getValue().toUpperCase(ENGLISH))))));
```

Comment:
Why ENGLISH?

Reply:
I don't know... That's what's used in code in similar situations. It's no longer recommended to not providing a locale.